src/Orleans.Transactions/State/TransactionalState.cs

```diff
@@ -80,7 +89,7 @@ public void Save()
             // Validation
             //
 
-            if (this.version.TransactionId > info.TransactionId || this.writeLowerBound >= info.TransactionId)
+            if (this.version.TransactionId > info.TransactionId || this.HighestRead >= info.TransactionId)
```

Comment:
Maybe I'm thinking about it the wrong way (and it's kinda early in the morning & pre-coffee) but would `HighestPersistedTransactionId` or `HighestWrittenTransactionId` be a clearer name? Maybe it's too verbose